crates/xmtp-networking/src/grpc_api_helper.rs

```diff
@@ -0,0 +1,99 @@
+use crate::proto_helper::xmtp::message_api::v1;
+
+use tonic::{metadata::MetadataValue, transport::Channel, Request};
+
+// Do a barebones unpaginated Query gRPC request
+// With optional PagingInfo
+pub async fn query(
+    host: String,
+    topic: String,
+    paging_info: Option<v1::PagingInfo>,
+) -> Result<v1::QueryResponse, tonic::Status> {
+    let mut client = v1::message_api_client::MessageApiClient::connect(host)
+        .await
+        .map_err(|e| tonic::Status::new(tonic::Code::Internal, format!("{}", e)))?;
+    let mut request = v1::QueryRequest {
+        content_topics: vec![topic],
+        ..Default::default()
+    };
+    // Check if paging_info is not None
+    if let Some(p) = paging_info {
+        request.paging_info = Some(p);
+    }
+    // Do the query and get a Tonic response that we need to process
+    let response = client.query(request).await;
+    response.map(|r| r.into_inner())
+}
+
+// Do a JSON serialized version of query, where the v1::QueryResponse is JSON serialized
+pub async fn query_serialized(
+    host: String,
+    topic: String,
+    json_paging_info: String,
+) -> Result<String, String> {
+    // Check if json_paging_info is not an empty string, if so deserialize it
+    let paging_info = if !json_paging_info.is_empty() {
+        let p: v1::PagingInfo =
+            serde_json::from_str(&json_paging_info).map_err(|e| format!("{}", e))?;
+        Some(p)
+    } else {
+        None
+    };
+
+    // Do the query and get a Tonic response that we need to process
+    let response = query(host, topic, paging_info)
+        .await
+        .map_err(|e| format!("{}", e))?;
+    // Response is a v1::QueryResponse protobuf message, which we need to serialize to JSON
+    let json = serde_json::to_string(&response).map_err(|e| format!("{}", e))?;
```

Comment:
How do the proto fields that are of type `bytes` make it through this JSON serialization?

Reply:
Great question, in my experience with a different Rust proto library they become base64-encoded.

Reply:
I lied, looks like they become byte arrays: "{\"content_topic\":\"\",\"timestamp_ns\":0,\"message\":[1,2,3,4]}"

Reply:
OK. That's great

Reply:
I think that's bad, Swift expects base64 from my test. I added in a hack until Prost becomes proto3 json compliant